app/policies/user_policy.rb

```diff
@@ -0,0 +1,42 @@
+class UserPolicy < ApplicationPolicy
+  class UserPolicy < ApplicationPolicy
+    def index?
+      user.admin?
+    end
+
+    def show?
+      user.admin? || record == user
+    end
+
+    def update?
+      user.admin? || record == user
+    end
+
+    def destroy?
+      user.admin? && user != record
+    end
+  end
+
+  class VehiclePolicy < ApplicationPolicy
```

Comment:
Why Vehicle Policy inside the User Policy file? The should be a separate file for Vechle